File: example2/file2words.py
import phish

def message_callback(message):
  for word in open(message, "r").read().split():
    phish.send((word, 1))

def closed_callback():
  phish.close()

phish.init()
phish.enable_input_port(0, message_callback, closed_callback)
phish.enable_output_port(0)
phish.check()
phish.loop()

File: minnow2/file2words.cpp
#include <fstream>
#include <iostream>
#include <phish.hpp>

void message_callback(int parts)
{
  const std::string filename = phish::unpack<std::string>();
  std::ifstream stream(filename.c_str());
  std::string word;
  while(true)
  {
    stream >> word;
    if(!stream)
      break;
    phish::pack(word);
    phish::pack(static_cast<uint32_t>(1));
    phish::send_key(word.c_str(), word.size());
  }
}

int main(int argc, char* argv[])
{
  phish::init(argc, argv);
  phish::input(0, message_callback, phish::exit);
  phish::output(0);
  phish::check();
  phish::loop();
}

